src/binaryen-c.cpp

```diff
@@ -1798,7 +1798,8 @@ void BinaryenSetFunctionTable(BinaryenModuleRef module, const char** funcNames,
     segment.data.push_back(funcNames[i]);
   }
   wasm->table.segments.push_back(segment);
-  wasm->table.initial = wasm->table.max = numFuncNames;
+  wasm->table.initial = initial = std::max(initial, numFuncNames);
```

Comment:
I'm not sure it's worth doing these attempted corrections - seems simpler to just apply the initial and max the user provided. If they are too small, the module won't validate, but that seems less surprising I think